internal/commands/database/create.go

```diff
@@ -0,0 +1,133 @@
+package database
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/commands"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/completion"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/config"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/labels"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/namedargs"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/output"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/ui"
+	"github.com/UpCloudLtd/upcloud-go-api/v8/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/v8/upcloud/request"
+	"github.com/spf13/pflag"
+)
+
+type createCommand struct {
+	*commands.BaseCommand
+	params createParams
+	wait   config.OptionalBoolean
+}
+
+func CreateCommand() commands.Command {
+	return &createCommand{
+		BaseCommand: commands.New("create", "Create a new database", "upctl database create"),
+	}
+}
+
+var defaultCreateParams = createParams{
+	CreateManagedDatabaseRequest: request.CreateManagedDatabaseRequest{
+		Plan: "2x2xCPU-4GB-100GB",
+		Type: upcloud.ManagedDatabaseServiceTypeMySQL,
+	},
+}
+
+type createParams struct {
+	request.CreateManagedDatabaseRequest
+	labels              []string
+	networks            []string
+	terminateProtection bool
+	dbType              string
+	properties          []string
+}
+
+func (s *createParams) processParams() error {
+	if len(s.labels) > 0 {
+		labelSlice, err := labels.StringsToSliceOfLabels(s.labels)
+		if err != nil {
+			return err
+		}
+		s.Labels = labelSlice
+	}
+
+	if s.terminateProtection {
+		s.TerminationProtection = &s.terminateProtection
+	}
+
+	if s.dbType != "" {
+		s.Type = upcloud.ManagedDatabaseServiceType(s.dbType)
+	}
+
+	if len(s.properties) > 0 {
+		var props request.ManagedDatabasePropertiesRequest
+		for _, prop := range s.properties {
+			parts := strings.SplitN(prop, "=", 2)
+			if len(parts) != 2 {
+				return fmt.Errorf("invalid property format: %s, expected key=value", prop)
+			}
+			key := upcloud.ManagedDatabasePropertyKey(parts[0])
+			props.Set(key, parts[1])
```

Comment:
We probably need some parsing (like you had in the first commit) here to have correct type for the value. There are some properties that take object or array as the value, so JSON unmarshal might be simplest solution even if the command would be a bit ugly.